### ISSUE TITLE
Pytorch 1.5 warning during to_fp16 fit

### ISSUE THREAD

**Issue by ProustIdee** [fastai/fastai#2563]
```
learn=language_model_learner(data_lm, pretrained=False, arch=AWD_LSTM, config=config).to_fp16()
learn.lr_find()
```

Produces the warning

`UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the gradient for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations.`

Looks like it landed end of January, between pytorch 1.4 and 1.5: https://github.com/pytorch/pytorch/pull/30531

`learn.fit_one_cycle(1, max_lr=1e-2, callbacks=[callbacks.SaveModelCallback(learn, every='epoch', monitor='accuracy', name='model')])`

Produces warnings that go on forever. Not sure if it actually is causing an issue with the training.

```text
=== Software ===
python        : 3.7.3
fastai        : 1.0.61
fastprogress  : 0.2.3
torch         : 1.5.0
torch cuda    : 10.2 / is available
torch cudnn   : 7605 / is enabled

=== Hardware ===
torch devices : 8
  - gpu0      : Tesla V100-SXM2-16GB
  - gpu1      : Tesla V100-SXM2-16GB
  - gpu2      : Tesla V100-SXM2-16GB
  - gpu3      : Tesla V100-SXM2-16GB
  - gpu4      : Tesla V100-SXM2-16GB
  - gpu5      : Tesla V100-SXM2-16GB
  - gpu6      : Tesla V100-SXM2-16GB
  - gpu7      : Tesla V100-SXM2-16GB

=== Environment ===
platform      : Linux-4.4.0-1075-aws-x86_64-with-debian-buster-sid
distro        : Ubuntu 18.04 bionic
conda env     : Unknown
python        : /usr/local/bin/python
sys.path      :
/usr/local/anaconda/lib/python37.zip
/usr/local/anaconda/lib/python3.7
/usr/local/anaconda/lib/python3.7/lib-dynload
/usr/local/anaconda/lib/python3.7/site-packages
/usr/local/anaconda/lib/python3.7/site-packages/IPython/extensions
no nvidia-smi is found
```

**Comment by sgugger** [comment]
The fix has been done in fastai v2. It requires deep changes in the `WeightDropout` layer of the AWD-LSTM so could have some backward-compatibility problems, so I would advise you to do one of the following:
- switch to the fastai2 model
- pin PyTorch to 1.4.0
- ignore all warnings

**Comment by ProustIdee** [comment]
Thanks sgugger! Options 2 and 3 both work.

I haven't tried fastai2, the documentation seems a bit sparser than fastai1, so I assumed it wasn't "final" yet.

**Comment by sgugger** [comment]
It's feature-complete but the doc is not yet entirely done indeed.

**Comment by aflip** [comment]
Hi, I am having the same issue.

I am not sure how to suppress the warning

I tried running

```
import warnings
warnings.filterwarnings("ignore", category=UserWarning, module="torch.nn.functional")
```
But this still shows the same error, although the training seems to be progressing.  I realized that I've written the wrong torch module, but which is the right module to add here? `torch.autograd.backward`?
